specs/core/0_beacon-chain.md

```diff
@@ -1217,14 +1208,14 @@ def get_initial_beacon_state(initial_validator_deposits: List[Deposit],
             poc_commitment=deposit.deposit_data.deposit_input.poc_commitment,
         )
         if get_effective_balance(state, validator_index) == MAX_DEPOSIT * GWEI_PER_ETH:
```

Comment:
This may be a bug. I think we want to process the deposits before the activations, in two separate loops. And for activations we want `>= MAX_DEPOSIT * GWEI_PER_ETH` (as opposed to `==`).

Reply:
What's the bug? I don't see it (the reason for two loops; I see the rationale for >= instead of ==).

Reply:
`get_effective_balance` can never be above `MAX_DEPOSIT * GWEI_PER_ETH` as is, but `==` is fragile if the internals of `get_effective_balance` change

Reply:
The bug (which I fixed before reading this) is that there could be multiple calls to `activate_validator`. This would create two entries in the delta hash chain.